Mathlib/CategoryTheory/Bicategory/LaxSlice.lean

```diff
@@ -0,0 +1,526 @@
+/-
+Copyright (c) 2025 Judah Towery. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Judah Towery
+-/
+module
+
+public import Mathlib.CategoryTheory.Bicategory.Functor.StrictPseudofunctor
+
+/-!
+
+# The lax slice bicategory F ↓ X of a lax functor F : B ⥤ᴸ C over an object X : C
+
+* objects are pairs (A : B, f_A : FA ⟶ X)
+* 1-cells are pairs (p : A₀ ⟶ A₁, θ_p : f₀ ⟶ f₁(Fp) in C
+* 2-cells are 2-cells α : p₀ ⟶ p₁ in B with Fα subject to the ice cream cone condition.
+
+Provides a change-of-slice strict pseudofunctor for a 1-cell u : X ⟶ Y,
+F ↓ u : (F ↓ X) ⥤ᵖ (F ↓ Y)
+
+## References
+* [Niles Johnson, Donald Yau, *2-Dimensional Categories*](https://arxiv.org/abs/2002.06055),
+section 7.1
+-/
+
+namespace CategoryTheory
+
+open Category Bicategory
+
+universe w₁ w₂ v₁ v₂
+
+namespace LaxSlice
+
+variable {B C : Type*} [Bicategory.{w₁, v₁} B] [Bicategory.{w₂, v₂} C] (F : B ⥤ᴸ C) (X : C)
+
+/-- Objects of the lax slice bicategory `F ↓ X`.
+A pair `(A, f_A)` where `A ∈ B` and `f_A : FA ⟶ X` in `C`. -/
+@[ext]
+structure Obj where
+  ob : B
+  map : F.obj ob ⟶ X
+
+/-- Notation for objects of lax slice bicategory. -/
+scoped notation F " ↓ " X => Obj F X
+
+/-- 1-cells in `F ↓ X`
+A 1-cell `(A₀, f₀) ⟶ (A₁, f₁)` is a pair `(p, θ_p)` with
+`p : A₀ ⟶ A₁` in `B`, and `θ_p : f₀ ⟶ f₁(Fp)` in `C`.
+This is depicted as a triangle
+```
+FA₀-----Fp----->FA₁
+|               |
+|    ⇒⇒θ_p⇒⇒    |
+|               |
+|--f₀-->X<--f₁--|
```

Comment:
I appreciate the effort to include diagrams in the docstrings! Could you try to draw actual triangles though, like 
```
FA₀ ---> FA₁
  \     /
   \   /
    v v
     X
```